src/sensors/SensorManager.cpp

```diff
@@ -36,110 +36,122 @@ namespace SlimeVR
-#elif IMU == IMU_BNO055
-                firstIMUAddress = I2CSCAN::pickDevice(0x29, 0x28, true);
-#elif IMU == IMU_MPU9250 || IMU == IMU_BMI160 || IMU == IMU_MPU6500 || IMU == IMU_MPU6050 || IMU == IMU_ICM20948
-                firstIMUAddress = I2CSCAN::pickDevice(0x68, 0x69, true);
-#else
-#error Unsupported primary IMU
-#endif
-
-                if (firstIMUAddress == 0)
-                {
-                    m_Sensor1 = new ErroneousSensor(0, IMU);
-                }
-                else
-                {
-                    m_Logger.trace("Primary IMU found at address 0x%02X", firstIMUAddress);
-
-#if IMU == IMU_BNO080 || IMU == IMU_BNO085 || IMU == IMU_BNO086
-                    m_Sensor1 = new BNO080Sensor(0, IMU, firstIMUAddress, IMU_ROTATION, PIN_IMU_INT);
-#elif IMU == IMU_BNO055
-                    m_Sensor1 = new BNO055Sensor(0, firstIMUAddress, IMU_ROTATION);
-#elif IMU == IMU_MPU9250
-                    m_Sensor1 = new MPU9250Sensor(0, firstIMUAddress, IMU_ROTATION);
-#elif IMU == IMU_BMI160
-                    m_Sensor1 = new BMI160Sensor(0, firstIMUAddress, IMU_ROTATION);
-#elif IMU == IMU_MPU6500 || IMU == IMU_MPU6050
-                    m_Sensor1 = new MPU6050Sensor(0, IMU, firstIMUAddress, IMU_ROTATION);
-#elif IMU == IMU_ICM20948
-                    m_Sensor1 = new ICM20948Sensor(0, firstIMUAddress, IMU_ROTATION);
-#endif
-                }
-
-                m_Sensor1->motionSetup();
-            }
+            bool foundIMU = false;
 
-            {
-#if SECOND_IMU == IMU_BNO080 || SECOND_IMU == IMU_BNO085 || SECOND_IMU == IMU_BNO086
-                secondIMUAddress = I2CSCAN::pickDevice(0x4B, 0x4A, false);
-#elif SECOND_IMU == IMU_BNO055
-                secondIMUAddress = I2CSCAN::pickDevice(0x28, 0x29, false);
-#elif SECOND_IMU == IMU_MPU9250 || SECOND_IMU == IMU_BMI160 || SECOND_IMU == IMU_MPU6500 || SECOND_IMU == IMU_MPU6050 || SECOND_IMU == IMU_ICM20948
-                secondIMUAddress = I2CSCAN::pickDevice(0x69, 0x68, false);
-#else
-#error Unsupported secondary IMU
-#endif
+            for (uint8_t i=0; i<IMU_COUNT; i+=2) {
+                uint8_t firstIMUAddress = 0;
+                uint8_t secondIMUAddress = 0;
+                Wire.begin(PIN_IMU_SDA, imuSCLPin[i/2]);
 
-                if (secondIMUAddress == firstIMUAddress)
                 {
-                    m_Logger.debug("No secondary IMU connected");
+                    if (imu[i] == IMU_BNO080 || imu[i] == IMU_BNO085 || imu[i] == IMU_BNO086)
+                        firstIMUAddress = I2CSCAN::pickDevice(0x4A, 0x4B, true);
+                    else if (imu[i] == IMU_BNO055)
+                        firstIMUAddress = I2CSCAN::pickDevice(0x29, 0x28, true);
+                    else if (imu[i] == IMU_MPU9250 || imu[i] == IMU_BMI160 || imu[i] == IMU_MPU6500 || imu[i] == IMU_MPU6050 || imu[i] == IMU_ICM20948)
+                        firstIMUAddress = I2CSCAN::pickDevice(0x68, 0x69, true);
```

Comment:
I assume the addresses here were swapped because of changes to `I2CSCAN::pickDevice`, but again, I'm confused of the purpose